src/squidpy/tl/_sliding_window.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import math
+from functools import reduce
+from typing import TYPE_CHECKING, Any, Literal, NamedTuple, Optional, Union
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from scanpy import logging as logg
+
+from squidpy._docs import d
+from squidpy._utils import NDArrayA
+from squidpy.gr._utils import _save_data
+
+__all__ = ["sliding_window"]
+
+
+@d.dedent
+def sliding_window(
+    adata: AnnData,
+    library_key: str | None = None,
+    window_size: int = 2500,
+    overlap: int = 0,
+    coord_columns: tuple[str, str] = ("globalX", "globalY"),
+    sliding_window_key: str = "sliding_window",
+    spatial_key: str = "spatial",
+    copy: bool = False,
+) -> AnnData:
+    """
+    Divide a tissue slice into regulary shaped spatially contiguous regions (windows).
+
+    Parameters
+        ----------
+        adata: AnnData
+            Annotated data matrix.
+        window_size: int
+            Size of the sliding window.
+        library_key: str
+            Only cells with the same library_key in `adata.obs` are assigned to the same window
+        coord_columns: Tuple[str, str]
+            Tuple of column names in `adata.obs` that specify the coordinates (x, y), e.i. ('globalX', 'globalY')
+        sliding_window_key: str
+            Base name for sliding window columns.
+        overlap: int
```

Comment:
What if `overlap` is negative? Is this intended to have spaced out sections? Also errorhandling

Reply:
What do you mean by spaced out sections? Sections that are not completely equally sized? Then yes

Reply:
Sorry, should've been clearer. If I were to set overlap to -10, I'd expect my windows to be 10 units apart each, instead of overlapping, right? Do we allow for this?